src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(name) {
+    if (name instanceof Herbivore) {
+      if (name.hidden === false) {
+        name.health -= 50;
+      }
+
+      if (name.health <= 0) {
+        const res = Animal.alive.filter(beast => beast.health > 0);
+
+        Animal.alive = res;
```

Comment:
we can do it in one line
```suggestion
        Animal.alive = Animal.alive.filter(beast => beast.health > 0);
```